keps/sig-storage/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,541 @@
+`patch` on PVC resource.
+
+### Test Plan
+
+1. Unit tests
+
+1. e2e tests
+    - RemoveOnScaleDown
+      1. Create 2 pod stateful set, scale to 1 pod, confirm PV deleted
+      1. Create 2 pod stateful set, add data to PVs, scale to 1 pod, scale back to 2, confirm PV empty
+      1. Create 2 pod stateful set, delete stateful set, confirm PVs deleted
+      1.Create 2 pod stateful set, add data to PVs, manually delete one pod, confirm pod comes back and PV has data (PV not deleted)
+      1. As above, but manually delete all pods in stateful set
+      1. Create 2 pod stateful set, add data to PVs, manually delete one pod, immediately scale down to one pod, confirm PV is deleted
+      1. Create 2 pod stateful set, add data to PVs, manually delete one pod, immediately scale down to one pod, scale back to two pods, confirm PV is empty
+    - RemoveOnStatefulSetDeletion
+      1. Create 2 pod stateful set, scale to 1 pod, confirm PV still exists
+      1. Create 2 pod stateful set, add data to PVs, scale to 1 pod, scale back to 2, confirm PV has data (PV not deleted)
+      1. Create 2 pod stateful set, delete stateful set, confirm PVs deleted
+      1. Create 2 pod stateful set, add data to PVs, manually delete one pod, confirm pod comes back and PV has data (PV not deleted)
+      1. As above, but manually delete all pods in stateful set
+      1. Create 2 pod stateful set, add data to PVs, manually delete one pod, immediately scale down to one pod, confirm PV exists
+      1. Create 2 pod stateful set, add data to PVs, manually delete one pod, immediately scale down to one pod, scale back to two pods, confirm PV has data
+    - Retain: 
+      1. same tests as above, but PVs not removed in any case
+
+1. Upgrade/Downgrade tests
+    1. Create statefulset in previous version and upgrade to the version 
+       supporting this feature. The PVCs should remain intact.
+    2. Downgrade to earlier version w
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).  Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+### Graduation Criteria
+
+#### Beta release
+- Complete adding the changes in the design.
+- Add unit, functional, upgrade and downgrade tests to automated k8s test.
+
+#### Alpha -> Beta Graduation
+
+- Gather feedback from developers and users via k8s github issues.
```

Comment:
You're only going by github issues?

Reply:
Thank you @kikisdeliveryservice  
Was only planning for GitHub issue feedback for now. Have not planned for a survey yet. Are there other ways you would recommend ?
Nevertheless do we need this since we are talking about getting to Alpha in this iteration ?

Reply:
>Was only planning for GitHub issue feedback for now. Have not planned for a survey yet. Are there other ways you would recommend ?

that would be up to you and your reviewers, but github issues seems like an insufficient metric

>Nevertheless do we need this since we are talking about getting to Alpha in this iteration ?
We enourage people to fill out as much as possible but for now alpha is only required.